deltacat/utils/ray_utils/concurrency.py

```diff
@@ -104,7 +104,13 @@ def round_robin_options_provider(
         foo.options(**opt).remote()
     ```
     """
-    assert resource_keys, f"No resource keys given to round robin!"
-    resource_key_index = i % len(resource_keys)
-    key = resource_keys[resource_key_index]
-    return {"resources": {key: resource_amount_provider(resource_key_index)}}
+    opts = kwargs.get("pg_config")
```

Comment:
Can we just add this as an additional named argument instead to make it easier to document going forward (e.g. add `pg_config` together with `resource_amount_provider` and `lambda i`)?